tests/test_auth.py

```
@@ -690,3 +690,36 @@ async def test_bearer_bad_token_rejected(self, no_cookie_client):
+    def test_session_user_empty_token_is_none(self, tmp_path):
+        mgr = AuthManager(tmp_path)
+        mgr.setup_user("alice", "Alice", "", "pw")
+        assert mgr.session_user("") is None
+
+    def test_get_user_bad_token_falls_back_to_first_user(self, tmp_path):
+        # Documents the footgun session_user avoids: get_user with a present
+        # but invalid token returns the first user, which is wrong for
+        # identity/ownership decisions.
+        mgr = AuthManager(tmp_path)
+        mgr.setup_user("alice", "Alice", "", "pw")
+        assert mgr.get_user(token="not-a-real-token") is not None
```

Ruff (0.15.15)

[error] Possible hardcoded password assigned to argument: "token"

(S106)